pkg/services/ngalert/notifier/channels/pushover.go

```diff
@@ -157,112 +167,129 @@ func (pn *PushoverNotifier) SendResolved() bool {
+		return nil, b, fmt.Errorf("failed to write the token: %w", err)
 	}
 
-	// Add priority
+	status := types.Alerts(as...).Status()
 	priority := pn.AlertingPriority
-	if alerts.Status() == model.AlertResolved {
+	if status == model.AlertResolved {
 		priority = pn.OKPriority
 	}
-	err = w.WriteField("priority", strconv.Itoa(priority))
-	if err != nil {
-		return nil, b, err
+	if err := w.WriteField("priority", strconv.Itoa(priority)); err != nil {
+		return nil, b, fmt.Errorf("failed to write the priority: %w", err)
 	}
 
 	if priority == 2 {
-		err = w.WriteField("retry", strconv.Itoa(pn.Retry))
-		if err != nil {
-			return nil, b, err
+		if err := w.WriteField("retry", strconv.Itoa(pn.Retry)); err != nil {
+			return nil, b, fmt.Errorf("failed to write retry: %w", err)
 		}
 
-		err = w.WriteField("expire", strconv.Itoa(pn.Expire))
-		if err != nil {
-			return nil, b, err
+		if err := w.WriteField("expire", strconv.Itoa(pn.Expire)); err != nil {
+			return nil, b, fmt.Errorf("failed to write expire: %w", err)
 		}
 	}
 
-	// Add device
 	if pn.Device != "" {
-		err = w.WriteField("device", tmpl(pn.Device))
-		if err != nil {
-			return nil, b, err
+		if err := w.WriteField("device", tmpl(pn.Device)); err != nil {
+			return nil, b, fmt.Errorf("failed to write the device: %w", err)
 		}
 	}
 
-	// Add sound
-	sound := tmpl(pn.AlertingSound)
-	if alerts.Status() == model.AlertResolved {
-		sound = tmpl(pn.OKSound)
-	}
-	if sound != "default" {
-		err = w.WriteField("sound", sound)
-		if err != nil {
-			return nil, b, err
-		}
+	if err := w.WriteField("title", tmpl(DefaultMessageTitleEmbed)); err != nil {
+		return nil, b, fmt.Errorf("failed to write the title: %w", err)
 	}
 
-	// Add title
-	err = w.WriteField("title", tmpl(DefaultMessageTitleEmbed))
-	if err != nil {
-		return nil, b, err
+	ruleURL := joinUrlPath(pn.tmpl.ExternalURL.String(), "/alerting/list", pn.log)
+	if err := w.WriteField("url", ruleURL); err != nil {
+		return nil, b, fmt.Errorf("failed to write the URL: %w", err)
 	}
 
-	// Add URL
-	err = w.WriteField("url", ruleURL)
-	if err != nil {
-		return nil, b, err
-	}
-	// Add URL title
-	err = w.WriteField("url_title", "Show alert rule")
-	if err != nil {
-		return nil, b, err
+	if err := w.WriteField("url_title", "Show alert rule"); err != nil {
+		return nil, b, fmt.Errorf("failed to write the URL title: %w", err)
 	}
 
-	// Add message
-	err = w.WriteField("message", tmpl(pn.Message))
-	if err != nil {
-		return nil, b, err
+	if err := w.WriteField("message", tmpl(pn.Message)); err != nil {
+		return nil, b, fmt.Errorf("failed write the message: %w", err)
 	}
 
-	if tmplErr != nil {
-		pn.log.Warn("failed to template pushover message", "err", tmplErr.Error())
+	// Pushover supports at most one image attachment with a maximum size of pushoverMaxFileSize.
+	// If the image is larger than pushoverMaxFileSize then return an error.
+	_ = withStoredImages(ctx, pn.log, pn.images, func(index int, image *ngmodels.Image) error {
```

Comment:
A similar consideration as I mentioned for Telegram: 

`withStoredImages` will short-circuit if `func(index int, image *ngmodels.Image)` returns an error. This means that if an image fails to send for whatever reason, no further images will attempt to send.